api/git/azure.go

```diff
@@ -241,13 +258,24 @@ func parseHttpUrl(rawUrl string) (*azureOptions, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to parse HTTP url")
 	}
-
+	azureDevOpsServerHost, azureDevOpsServerHostOK := os.LookupEnv("AZURE_DEVOPS_SERVER_HOST")
+	azureDevOpsServerBaseUrl, azureDevOpsServerBaseUrlOK := os.LookupEnv("AZURE_DEVOPS_SERVER_BASE_URL")
 	opt := azureOptions{}
 	switch {
-	case u.Host == azureDevOpsHost:
+	case (u.Host == azureDevOpsHost) || (azureDevOpsServerHostOK && u.Host == azureDevOpsServerHost):
 		path := strings.Split(u.Path, "/")
+		if azureDevOpsServerBaseUrlOK {
+			path = strings.Split(strings.ReplaceAll(u.Path, azureDevOpsServerBaseUrl+"/", ""), "/")
+		}
 		if len(path) != 5 {
-			return nil, errors.Errorf("want url %s, got %s", expectedAzureDevOpsHttpUrl, u)
+			expectedUrl := expectedAzureDevOpsHttpUrl
+			if azureDevOpsServerHostOK && azureDevOpsServerBaseUrlOK {
+				expectedUrl = "https://" + azureDevOpsServerHost + "/" + azureDevOpsServerBaseUrl + "/Organisation/Project/_git/Repository"
+			}
+			if azureDevOpsServerHostOK {
+				expectedUrl = "https://" + azureDevOpsServerHost + "/Organisation/Project/_git/Repository"
```

Comment:
afaik the `/Organization/Project` is not part of the AzureDevOps Server url; it is usually in form of `http://ServerName/{DefaultCollection}/{ProjectName}`

So for example clone url: `https://some_server/tfs/SomeCollectionName/SomeProjectName/_git/GitRepoName`


see https://learn.microsoft.com/en-us/azure/devops/organizations/projects/connect-to-projects?toc=%2Fazure%2Fdevops%2Frepos%2Ftoc.json&view=azure-devops-2022&tabs=visual-studio-2022

If this uses AzureDevOps rest api then the url is in the form of: 
`GET https://{instance}/{collection}/{project}/_apis/git/repositories/{repositoryId}?api-version=7.0`

https://learn.microsoft.com/en-us/rest/api/azure/devops/git/commits/get?view=azure-devops-server-rest-7.0&tabs=HTTP